pkg/cluster/cluster.go

```diff
@@ -159,7 +159,18 @@ func (c *Cluster) prepareBackupAndRestore() error {
 			}
 		}
 	}
-	err := k8sutil.CreateBackupReplicaSetAndService(c.KubeCli, c.Name, c.Namespace, c.spec.Backup)
+
+	podSpec, err := k8sutil.MakeBackupPodSpec(c.Name, c.spec.Backup)
+	if err != nil {
+		return err
+	}
+	switch backup.StorageType {
```

Comment:
hmm... there are too many places switch around storage Type.

I think we should probably make that switch top level.